crates/table/src/bflatn_to_bsatn_fast_path.rs

```diff
@@ -244,8 +244,8 @@ impl LayoutBuilder {
         // but BSATN stores `(tag, payload)`,
         // then splice the `first_variant_layout` into `self`.
 
-        let payload_bflatn_offset = self.next_bflatn_offset();
-        let tag_bflatn_offset = payload_bflatn_offset + sum.tag_offset;
+        let tag_bflatn_offset = self.next_bflatn_offset();
+        let payload_bflatn_offset = tag_bflatn_offset + sum.payload_offset;
```

Comment:
I disagree - it should be easy to change the `visit_sum` code to do a `visit_primitive(&PrimitiveType::U8)` for the tag, and then to advance to the payload and add the payload fields using logic similar to `visit_product`. I think this will be simpler than adding an additional pass, and I'd prefer not to add a post-processing step to handle a single case as it seems likely to go forgotten if this code ever changes.

Reply:
Certainly as a first step we should simply replace the `self.fields.push(...)` on line 253 with a call to `visit_primitive(&PrimitiveType::U8)`.